libraries/botbuilder-dialogs-adaptive/src/skills/adaptiveSkillDialog.ts

```diff
@@ -0,0 +1,131 @@
+
+    /**
+     * The Microsoft App ID that will be calling the skill.
+     * 
+     * @remarks
+     * Defauls to a value of `=settings.MicrosoftAppId` which retrievs the bots ID from settings.
+     */
+    public botId = new StringExpression('=settings.MicrosoftAppId');
+
+    /**
+     * The callback Url for the skill host.
+     * 
+     * @remarks
+     * Defauls to a value of `=settings.SkillHostEndpoint` which retrieves the endpoint from settings.
+     */
+    public skillHostEndpoint = new StringExpression('=settings.SkillHostEndpoint');
+
+    /**
+     * The Microsoft App ID for the skill.
+     */
+    public skillAppId: StringExpression;
+
+    /**
+     * The `/api/messages` endpoint for the skill.
+     */
+    public skillEndpoint: StringExpression;
+
+    /**
+     * Template for the activity.
+     */
+    public activity: TemplateInterface<Partial<Activity>>
+
+    /**
+     * Creates a new `AdaptiveSkillDialog instance.
+     * @param options Optional options used to configure the skill dialog.
+     */
+    constructor(options?: SkillDialogOptions) {
+        super(Object.assign({ skill: {} } as SkillDialogOptions, options));
+    }
+
+    public async beginDialog(dc: DialogContext, options: Partial<BeginSkillDialogOptions>): Promise<DialogTurnResult> {
+        const dcState = dc.state;
+        if (this.disabled && this.disabled.getValue(dcState)) {
+            return await dc.endDialog();
+        }
+
+        // Update options with global options
+        const globalOptions = dc.context.turnState.get(GLOBAL_SKILL_OPTIONS);
+        if (globalOptions) {
+               this.dialogOptions = Object.assign(globalOptions, this.dialogOptions);
+        }
+        if (!this.dialogOptions.conversationState) { dc.dialogManager.conversationState }
+
+        // Setup the skill to call
+        const botId = this.botId.getValue(dcState);
+        const skillHostEndpoint = this.skillHostEndpoint.getValue(dcState);
+        if (botId) { this.dialogOptions.botId = botId }
+        if (skillHostEndpoint) { this.dialogOptions.skillHostEndpoint = skillHostEndpoint }
+        if (this.skillAppId) { this.dialogOptions.skill.id = this.dialogOptions.skill.appId = this.skillAppId.getValue(dcState) }
+        if (this.skillEndpoint) { this.dialogOptions.skill.skillEndpoint = this.skillEndpoint.getValue(dcState) }
+
+        // Get the activity to send to the skill.
+        options = {}
+        if (this.activityProcessed.getValue(dcState) && this.activity) {
+            // The parent consumed the activity in context, use the Activity property to start the skill.
+            options.activity = await this.activity.bindToData(dc.context, dcState);
+        } else {
+            // Send the turn context activity to the skill (pass through).
+            options.activity = dc.context.activity;
+        }
+
+        // Call the base to invoke the skill
+        return await super.beginDialog(dc, options);
```

Comment:
Is the [connectionName](https://github.com/microsoft/botbuilder-js/blob/9894c150b3a97d4ea772380769e6083bfbd99297/libraries/botbuilder-dialogs/src/beginSkillDialogOptions.ts#L23) passed in through this code? It looks like the passed-in `options` is entirely discarded.

This potential issue also exists in C#: [AdaptiveSkillDialog.cs#L138](https://github.com/microsoft/botbuilder-dotnet/blob/6adc879925e2f8c3f19804b21d8adab7b82098a1/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs#L138)

Reply:
@stevenic, @gabog, @swagatmishra2007

Reply:
resolved as of https://github.com/microsoft/botbuilder-js/pull/2177/commits/f443f0be3521b2b9494e91055a50ef8ddaa231cd